swap_meet/vendor.py

```diff
@@ -0,0 +1,101 @@
+class Vendor: 
+    def __init__(self,inventory = None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self,item):
+    #Not sure if this is the best use of the try and except clause, wanted to attempt as practice. Alternatively, I could have written: 
+    #if item not in self.inventory: return False else: self.inventory.remove(item) return item. 
+        try:
+            self.inventory.remove(item)
+            return item
+        except ValueError:
+            return False
+    
+    def get_by_category(self,category):
+        result = []
+        for item in self.inventory: 
+            if item.category == category: 
+                result.append(item)
+        return result 
+
+
+    def swap_items(self,friend_vendor,my_item,their_item): 
+        if (my_item in self.inventory and
+                their_item in friend_vendor.inventory):
+            self.inventory.remove(my_item)
+            friend_vendor.inventory.append(my_item)
+            friend_vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            return True
+        else:
+            return False
+
+    def swap_first_item(self,friend_vendor):
+        if len(self.inventory) == 0 or len(friend_vendor.inventory) == 0:
+            return False
+        
+        self.inventory.append(friend_vendor.inventory[0])
+        friend_vendor.inventory.append(self.inventory[0])
+        self.inventory.remove(self.inventory[0])
+        friend_vendor.inventory.remove(friend_vendor.inventory[0])
```

Comment:
You can DRY this up a little using `swap_items`:

```suggestion
        self.inventory.swap_items(friend_vendor, self.inventory[0], friend_vendor.inventory[0])
```